### PR TITLE
AP_RPM: Giving RPM some TLC

### PR DESCRIPTION
**Pull request by Gone4Dirt**
This is a minor fix to what I believe is just a hang over.

It looks like we create param arrays for _maximum, _minimum, and _quality_min yet we never give users the chance to set them on instances other than the 1st.  This doesn't have any ploblems until we start trying to add more than two instances of RPM and immediatly follow on the param idx.  Only noticed it whilst helping out @GiovannaAF trying to add more instances to RPM.

So we should either reduce the variable array size or add the _maximum, _minimum, and _quality_min params to the >1 instances of RPM IMHO.  I opten for the former here:

### CODE REVIEW

**Review by tridge — CHANGES_REQUESTED** [review]
as discussed on call, make 2_TYPE an enable param and add extra params for the 2nd instance

**Comment by Gone4Dirt** [comment]
Recommend reviewing per commit as there are a few things in this PR.

This PR now does the following:
- Fixes the bug where by users cannot access _MIN and _Max params.
- Gives each instance of RPM a _MIN_QUAL param.
- Re-factors the params in such a way so as to make it easier to expand the number of instances (I know of a few people wanting to do this)
- Removes type = PWM as this is a historic hangover from the PX4 split.  I need to do the param conversion for the above point anyway so I figure this is the perfect time to do it.
- Minor whitespace tidy up.

File: libraries/AP_RPM/AP_RPM.cpp
Comment (tridge) [inline libraries/AP_RPM/AP_RPM.cpp:135]: maybe use find_top_level_key_by_pointer()

File: libraries/AP_RPM/AP_RPM.cpp
Comment (tridge) [inline libraries/AP_RPM/AP_RPM.cpp:146]: skip the rest if we type was not set?

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_RPM/AP_RPM.cpp
Comment (Gone4Dirt) [reply libraries/AP_RPM/AP_RPM.cpp]: Done and tested.  Confirmed working.

File: libraries/AP_RPM/AP_RPM.cpp
Comment (Gone4Dirt) [reply libraries/AP_RPM/AP_RPM.cpp:155]: Done and tested.  Confirmed working.

**Comment by Gone4Dirt** [comment]
Wiki update here: https://github.com/ArduPilot/ardupilot_wiki/pull/3701

**Comment by Gone4Dirt** [comment]
My apologies, I cannot attend the dev call but I think this PR is good to go.  I would appriciate a review/potential merge.

**Review by tridge — APPROVED** [review]

File: libraries/AP_RPM/AP_RPM.h
Comment (tridge) [inline libraries/AP_RPM/AP_RPM.h:41]: better to leave the old numbers, and still map PWM to PIN, but don't list PWM in the docs. This makes it better if users manually copy params or use older docs

File: libraries/AP_RPM/AP_RPM.cpp
Comment (peterbarker) [inline libraries/AP_RPM/AP_RPM.cpp:100]: We might need to be more conservative with this

File: libraries/AP_RPM/AP_RPM.h
Comment (rmackay9) [inline libraries/AP_RPM/AP_RPM.h:42]: Let's keep the enumeration the same as before.  So if "PWM" isn't a valid number anymore, let's just remove "1" as a valid option and start from "2"

File: libraries/AP_RPM/AP_RPM.cpp
Comment (tridge) [inline libraries/AP_RPM/AP_RPM.cpp:100]: discussed on call, better to put year and month when it went in (so August 2021)

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_RPM/AP_RPM.cpp
Comment (peterbarker) [reply libraries/AP_RPM/AP_RPM.cpp:100]: Maybe a single line so grep shows us all `PARAMETER_CONVERSION` information?

**Comment by Gone4Dirt** [comment]
@Hwurzburg i have already have a wiki pr to reflect these code changes.

**Comment by Hwurzburg** [comment]
I will make it delay merge until this gets merged...and you make the minor edits ...thanks

**Comment by Gone4Dirt** [comment]
Updated the param conversion note.  Should be good to go once checks passed.

**Review by IamPete1 — APPROVED** [review]
LGTM

**Review by tridge — APPROVED** [review]

**Pull request merged**
